openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIResponseMapper.java

```diff
@@ -884,6 +933,58 @@ private ApiResponses handleRecordTypeSymbol(String referenceName, Map<String, Sc
         return apiResponses;
     }
 
+    /**
+     * Create API responses when return type is type reference.
+     */
+    private ApiResponses createResponseForPrimitiveTypeReferenceType(String referenceName,
+                                                                     TypeSymbol referredTypeSymbol,
+                                                                     TypeReferenceTypeSymbol typeReferenceTypeSymbol,
+                                                                     Map<String, Schema> schema,
+                                                                     Optional<String> customMediaPrefix,
+
+                                                                    OpenAPIComponentMapper componentMapper,
+                                                                    Map<String, Header> headers) {
+
+        TypeDescKind typeDescKind = referredTypeSymbol.typeKind();
+        io.swagger.v3.oas.models.media.MediaType media = new io.swagger.v3.oas.models.media.MediaType();
+
+        String statusCode = httpMethod.equals(POST) ? HTTP_201 : HTTP_200;
+        String description = httpMethod.equals(POST) ? HTTP_201_DESCRIPTION : HTTP_200_DESCRIPTION;
+
+        ApiResponses apiResponses = new ApiResponses();
+
+        if (typeDescKind == TypeDescKind.INTERSECTION) {
+            List<TypeSymbol> typeSymbols = ((IntersectionTypeSymbol) referredTypeSymbol).memberTypeDescriptors();
+            for (TypeSymbol symbol: typeSymbols) {
+                if (!(symbol instanceof ReadonlyTypeSymbol)) {
+                    typeDescKind = symbol.typeKind();
+                    break;
+                }
+            }
```

Comment:
Shall we move this logic to a separate function since we are using this in other places as well?

Reply:
Added. Please review